packages/expo-image/android/src/main/java/expo/modules/image/Image.kt

```diff
@@ -1,6 +1,18 @@
 package expo.modules.image
 
+import android.graphics.drawable.BitmapDrawable
 import android.graphics.drawable.Drawable
 import expo.modules.kotlin.sharedobjects.SharedRef
 
-class Image(ref: Drawable) : SharedRef<Drawable>(ref)
+class Image(ref: Drawable) : SharedRef<Drawable>(ref) {
+  override fun getAdditionalMemoryPressure(): Int {
+    val ref = ref
+    if (ref is BitmapDrawable) {
+      return ref.getBitmap().allocationByteCount
+    }
+
+    // We can't get the size in bytes of the drawable.
+    // Let's just return the size in pixels for now.
+    return ref.intrinsicWidth * ref.intrinsicHeight
```

Comment:
Maybe it makes sense to add some constant here to better approximate it? An RGBA pixel is probably 4 bytes, but can drawables have compression?

Reply:
If it's an image-like asset, it should be handled by the first if. Here, we're more likely dealing with animated assets and vector-like formats. I'll try to improve that function in the future, but firstly, I need to see how many different versions of drawable glide are producing.